app/src/main/java/com/guruthedev/instagram/viewModel/LoginViewModel.kt

```diff
@@ -22,6 +21,10 @@ class LoginViewModel : ViewModel() {
     val errorLiveData: LiveData<LoginError>
         get() = _errorLiveData
 
+    init {
+        preference = SharedPreference(application.applicationContext)
```

Comment:
can we remove this initialisation from here and do it in the Application class, the reason behind this is we dont need to initialise the preference in every class where the preference is used. Also this will reduce code repetition.